### DISCUSSION TITLE
example "package.json" not working: dependency not viable: "@theia/api-samples"

### DISCUSSION THREAD

**Discussion by talbym** [eclipse-theia/theia#8371] [General] [answered]
In the provided example package.json file for building a Theia IDE, there is a dependency which cannot be resolved as of 13th August 2020. The file is located here:
[https://github.com/eclipse-theia/theia/blob/master/examples/browser/package.json](https://github.com/eclipse-theia/theia/blob/master/examples/browser/package.json)
The defined dependency in question is listed on line 17:
"@theia/api-samples": "^1.4.0"
There is no entry for this extension here:
[https://www.npmjs.com/search?q=keywords%3Atheia-extension](https://www.npmjs.com/search?q=keywords%3Atheia-extension)
Thus, the build is not possible unless this line becomes deleted in the package.json file - or the extension would be added to the npmjs registry.

**Comment by kittaakos** [comment, marked as answer]
Thanks for your interest. This is not a bug, @talbym. The `@theia/api-samples` package is private and was created to provide examples for Theia-based apps started from the source. As far as I know, we have no intention of publishing it. You can use the `api-sample` to check how to use the APIs.

If you want to create an app for your self, jo can check out https://github.com/theia-ide/theia-apps how to define your dependencies.

**Comment by talbym** [comment]
@kittaakos Thanks for your reply. Although You do not consider line 17 a bug, the removal of it would still help beginner developers to approach Theia!
Myself being a beginner developer appraoching Theia, I followed this documentation page for developers:
[https://github.com/eclipse-theia/theia/blob/master/doc/Developing.md](https://github.com/eclipse-theia/theia/blob/master/doc/Developing.md)
There it is stated: "This file contains tips to help you take (and understand) your first steps in the world of Theia development. (...) How to build Theia and the example applications". On this page there is then a section which places a link to the example package.json file in question, the section being this one:
[https://github.com/eclipse-theia/theia/blob/master/doc/Developing.md#run-the-browser-based-example-application](https://github.com/eclipse-theia/theia/blob/master/doc/Developing.md#run-the-browser-based-example-application)
I understood that linking in this section to the example package.json file would be a recommendation for using it for a first try to build a Theia IDE, which is a great and highly appreciated offer. Therefore, removing line 17 from the example file, especially as the example file is referenced in the documention and this enrichment of the documentation being of great value, would be very helpful for beginning developers approaching Theia.

**Comment by akosyakov** [comment]
> Therefore, removing line 17 from the example file, especially as the example file is referenced in the documention and this enrichment of the documentation being of great value, would be very helpful for beginning developers approaching Theia.

I don't get it? Why should we remove it? Examples applications in this repo only for testing purposes of contributors, not to be used in production. If you want to build your app you should not follow contributor development docs, but application developer docs: https://theia-ide.org/docs/composing_applications/

**Reply by akosyakov** [reply]
> I understood that linking in this section to the example package.json file would be a recommendation for using it for a first try to build a Theia IDE, which is a great and highly appreciated offer.

You should follow https://github.com/eclipse-theia/theia/blob/master/doc/Developing.md#quick-start to build from sources.

**Comment by talbym** [comment]
I see what happened. I have been starting my journey on the documentation page which you (@akosyakov) recommended ([https://theia-ide.org/docs/composing_applications/](https://theia-ide.org/docs/composing_applications/)) me to follow, and there I followed the link to read about the [Prerequisites](https://github.com/eclipse-theia/theia/blob/master/doc/Developing.md#prerequisites) for building my own IDE, which actually is a link to the github page which I then studied. Further down on the github page is then the link to this (really nice) example file and I thought this would be an example targeted to the public, targeted to app developers.
I will think about how I might be able to provide you with a suggestion for improving the documentation a little bit on the [https://theia-ide.org/docs/composing_applications/](https://theia-ide.org/docs/composing_applications/) page, then. Being a Theia and also a Node.js beginner, I should suggest adding some information to the page which for you experts is not seen necessary to mention no more. Some extra information could have shortened my journey to my first own IDE, and could als have avoided the misunderstanding. I will take my time and think about it and then report back!
So far, thanks a lot for Theia!!

**Comment by vince-fugnitto** [comment]
@talbym each extension is documented at a high-level, including `@theia/api-samples`. When building applications, application-developers should understand which extensions to include to suit their use-cases and needs.

The `@theia/api-samples` extension is `private` (not-published to the `npm registry`), and it's [documentation](https://github.com/eclipse-theia/theia/blob/master/examples/api-samples/README.md) states:

> The extension is for reference and test purposes only and is not published on npm (private: true).

I hope this makes it clearer. If you are looking for examples of applications please see the examples found on [theia-apps](https://github.com/theia-ide/theia-apps).
